test-exports-cjs/index.mjs

```diff
@@ -0,0 +1,46 @@
+import assert from "assert";
+import { OpenAI } from "langchain";
+import { loadPrompt } from "langchain/prompts";
+import { HNSWLib } from "langchain/vectorstores";
+import { OpenAIEmbeddings } from "langchain/embeddings";
+import { InMemoryDocstore, Document } from "langchain/docstore";
+import { CSVLoader } from "langchain/document_loaders";
+
+// Test exports
+assert(typeof OpenAI === "function");
+assert(typeof loadPrompt === "function");
+assert(typeof HNSWLib === "function");
+
+// Test dynamic imports of peer dependencies
+const { HierarchicalNSW } = await HNSWLib.imports();
```

Comment:
this line surprised me; I'm not used to seeing the type of thing happening at runtime. I know you're just testing here, but the fact that `imports()` is part of the public api made me glance twice.

dealing w/ deps / peer deps / optional peer deps is really tricky as I'm sure you're well aware, especially across diff environments.

for now, if this abstraction works for you, I'd say run with it. just wanted to point out that it was surprising

Reply:
Hmm yea not without its issues, but it has mostly worked for us. Surprisingly hard to do optional deps in JS nowadays